Reflection free serializers ArrayIndexOutOfBoundsException

### Describe the bug

When updating from version 3.15.1 to 3.16.2, the build fails due to the property:

`quarkus.rest.jackson.optimization.enable-reflection-free-serializers=true.`

In JacksonCodeGenerator at line 284, the check passes, but `set` is the name of a field rather than a method prefix.

`record ExampleRequestBody(boolean required, boolean set) {}`


<details><summary>Stack Trace</summary>
<p>

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
        at org.jboss.jandex.MethodInfo.parameterType(MethodInfo.java:180)
        at io.quarkus.resteasy.reactive.jackson.deployment.processor.JacksonCodeGenerator$FieldSpecs.fieldType(JacksonCodeGenerator.java:285)
        at io.quarkus.resteasy.reactive.jackson.deployment.processor.JacksonCodeGenerator$FieldSpecs.<init>(JacksonCodeGenerator.java:271)
        at io.quarkus.resteasy.reactive.jackson.deployment.processor.JacksonCodeGenerator.fieldSpecsFromField(JacksonCodeGenerator.java:235)
        at io.quarkus.resteasy.reactive.jackson.deployment.processor.JacksonSerializerFactory.serializeFields(JacksonSerializerFactory.java:224)
        at io.quarkus.resteasy.reactive.jackson.deployment.processor.JacksonSerializerFactory.serializeObjectData(JacksonSerializerFactory.java:217)
        at io.quarkus.resteasy.reactive.jacployment.processor.JacksonSerializerFactory.create(JacksonSerializerFactory.java:138)actory.java:200)
        at io.quarkus.resteasy.reactive.jackson.deployment.processor.ResteasyReactiveJacksonProcessor.handleEndpointParams(ResteasyReactiveJacksonProcessor.java:419)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at java.base/java.lang.Thread.run(Thread.java:1575)
        at org.jboss.threads.JBossThread.run(JBossThread.java:499)

```

</p>
</details> 

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "23.0.1" 2024-10-15 OpenJDK Runtime Environment Homebrew (build 23.0.1) OpenJDK 64-Bit Server VM Homebrew (build 23.0.1, mixed mode, sharing)

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonCodeGenerator.java
+++ b/extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonCodeGenerator.java
@@ -161,7 +161,7 @@ public abstract class JacksonCodeGenerator {
         MAP(true),
         TYPE_VARIABLE(true);
 
-        private boolean generic;
+        private final boolean generic;
 
         FieldKind(boolean generic) {
             this.generic = generic;
@@ -281,7 +281,7 @@ public abstract class JacksonCodeGenerator {
             if (isPublicField()) {
                 return fieldInfo.type();
             }
-            if (methodInfo.name().startsWith("set")) {
+            if (methodInfo.parametersCount() == 1 && methodInfo.name().startsWith("set")) {
                 return methodInfo.parameterType(0);
             }
             return methodInfo.returnType();
@@ -304,6 +304,9 @@ public abstract class JacksonCodeGenerator {
 
         private String fieldNameFromMethod(MethodInfo methodInfo) {
             String methodName = methodInfo.name();
+            if (methodName.equals("get") || methodName.equals("set") || methodName.equals("is")) {
+                return methodName;
+            }
             if (methodName.startsWith("is")) {
                 return methodName.substring(2, 3).toLowerCase() + methodName.substring(3);
             }

--- a/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/FieldNameSetGetPrefixResourceTest.java
+++ b/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/FieldNameSetGetPrefixResourceTest.java
@@ -1,0 +1,52 @@
+package io.quarkus.resteasy.reactive.jackson.deployment.test;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+import io.smallrye.common.annotation.NonBlocking;
+
+// Ensures uncommon field names like "set", "get", and "is" are generated correctly.
+class FieldNameSetGetPrefixResourceTest {
+    @RegisterExtension
+    static QuarkusUnitTest test = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(Resource.class, Resource.UncommonBody.class).addAsResource(
+                            new StringAsset(
+                                    "quarkus.rest.jackson.optimization.enable-reflection-free-serializers=true\n"),
+                            "application.properties"));
+
+    @Test
+    void testFieldNameSetGetIsPrefix() {
+        RestAssured.get("/field-name-prefixes")
+                .then()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("id", Matchers.equalTo("id"))
+                .body("set", Matchers.is(true))
+                .body("get", Matchers.is(true))
+                .body("is", Matchers.is(false))
+                .body("setText", Matchers.equalTo("setText"));
+    }
+
+    @NonBlocking
+    @Path("/field-name-prefixes")
+    private static class Resource {
+        @GET
+        public UncommonBody get() {
+            return new UncommonBody("id", true, true, false, "setText");
+        }
+
+        private record UncommonBody(String id, boolean set, boolean get, boolean is, String setText) {
+        }
+    }
+
+}

